source/grid/grid_tools.cc

```diff
@@ -850,51 +850,115 @@ namespace GridTools
+    // compare points that have the same x coordinate when we try to find
+    // duplicated vertices.
+
+    // Start by finding the longest coordinate direction. This minimizes the
+    // number of points that need to be compared against each-other in a
+    // single set for typical geometries.
+    Point<spacedim> min;
+    std::fill(min.begin_raw(),
+              min.end_raw(),
+              std::numeric_limits<double>::max());
+    Point<spacedim> max;
+    std::fill(max.begin_raw(),
+              max.end_raw(),
+              std::numeric_limits<double>::lowest());
+    for (const Point<spacedim> &vertex : vertices)
+      for (unsigned int d = 0; d < spacedim; ++d)
+        {
+          min[d] = std::min(min[d], vertex[d]);
+          max[d] = std::min(max[d], vertex[d]);
+        }
+    unsigned int longest_coordinate_direction = 0;
+    double       longest_coordinate_length    = max[0] - min[0];
+    for (unsigned int d = 1; d < spacedim; ++d)
+      if (longest_coordinate_length < max[d] - min[d])
+        {
+          longest_coordinate_length    = max[d] - min[d];
+          longest_coordinate_direction = d;
+        }
 
-    // now loop over all vertices to be
-    // considered and try to find an identical
-    // one
-    for (unsigned int i = 0; i < considered_vertices.size(); ++i)
+    // Sort vertices (while preserving their vertex numbers) along that
+    // coordinate direction:
+    std::vector<std::pair<unsigned int, Point<spacedim>>> sorted_vertices;
+    sorted_vertices.reserve(vertices.size());
+    for (const unsigned int vertex_n : considered_vertices)
       {
-        Assert(considered_vertices[i] < vertices.size(), ExcInternalError());
-        if (new_vertex_numbers[considered_vertices[i]] !=
-            considered_vertices[i])
-          // this vertex has been identified with
-          // another one already, skip it in the
-          // test
-          continue;
-        // this vertex is not identified with
-        // another one so far. search in the list
-        // of remaining vertices. if a duplicate
-        // vertex is found, set the new vertex
-        // index for that vertex to this vertex'
-        // index.
-        for (unsigned int j = i + 1; j < considered_vertices.size(); ++j)
+        AssertIndexRange(vertex_n, vertices.size());
+        sorted_vertices.emplace_back(vertex_n, vertices[vertex_n]);
+      }
+    std::sort(sorted_vertices.begin(),
+              sorted_vertices.end(),
+              [&](const std::pair<unsigned int, Point<spacedim>> &a,
+                  const std::pair<unsigned int, Point<spacedim>> &b) {
+                return a.second[longest_coordinate_direction] <
+                       b.second[longest_coordinate_direction];
+              });
+
+    auto within_tolerance = [=](const Point<spacedim> &a,
+                                const Point<spacedim> &b) {
+      for (unsigned int d = 0; d < spacedim; ++d)
+        if (tol < std::abs(a[d] - b[d]))
```

Comment:
I prefer to use `<` everywhere since it makes more sense to me to have the smaller item on the left. This issue is not at all significant but I think it would be worthwhile to have a correct answer to this in the style guide so that we are internally consistent. Does anyone care enough about the issue to chime in?

Reply:
I think you two have a stronger preference than I do so I will change it.